gbpservice/neutron/nsf/core/threadpool.py

```diff
@@ -0,0 +1,80 @@
+import time
+import eventlet
+eventlet.monkey_patch()
+from eventlet import event
+from eventlet import greenpool
+from eventlet import greenthread
+import os
+import sys
+import threading
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+def _thread_done(gt, *args, **kwargs):
+    kwargs['pool'].thread_done(kwargs['thread'])
+
+
+class Thread(object):
+
+    def __init__(self, thread, pool):
+        self.thread = thread
+        self.thread.link(_thread_done, pool=pool, thread=self)
+
+    def stop(self):
```

Comment:
these are available for callers to invoke. Right now there is no condition where we want to stop thread in between. It finishes completely.